src/wp-includes/option.php

```diff
@@ -253,6 +253,74 @@ function get_option( $option, $default_value = false ) {
 	return apply_filters( "option_{$option}", maybe_unserialize( $value ), $option );
 }
 
+/**
+ * Prime specific options into the cache with a single database query.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array $options An array of option names to be primed.
+ */
+function prime_options( $options ) {
+	$alloptions       = wp_load_alloptions();
+	$options_to_prime = array_diff( $options, array_keys( $alloptions ) );
+
+	if ( ! empty( $options_to_prime ) ) {
+		global $wpdb;
+		$option_names = implode( "','", array_map( 'esc_sql', $options_to_prime ) );
+		$query   = "SELECT option_name, option_value FROM {$wpdb->options} WHERE option_name IN ('$option_names')";
+		$results = $wpdb->get_results( $query, OBJECT_K );
+
+		foreach ( $results as $result ) {
+			$alloptions[ $result->option_name ] = maybe_unserialize( $result->option_value );
+		}
+
+		wp_cache_set( 'alloptions', $alloptions, 'options' );
```

Comment:
These values must not be populated into `alloptions`, but individual cache values. So I would recommend you fill a new array `$option_values` with option names and their values and then pass it to `wp_cache_set_multiple( $option_values, 'options' )` at the end.

Reply:
This still stores values in `alloptions`, but it needs to be removed. Only the individual caches like you have below should be set by this. If the option wasn't already in `alloptions`, it shouldn't be stored in `alloptions` either.

Reply:
Sorry i forgot to remove it 😞